src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php

```diff
@@ -721,38 +724,89 @@ public static function analyze(
+                                    . "\"{$parent_const_storage->type->getId()}\" inherited from "
+                                    . "{$parent_classlike_storage->name}::{$const_name}",
+                                $type_location,
+                                "{$class_storage->name}::{$const_name}"
+                            ),
+                            $const_storage->suppressed_issues
+                        );
+                    } else {
+                        // Completely different
+                        IssueBuffer::maybeAdd(
+                            new InvalidClassConstantType(
+                                "The type \"{$const_storage->type->getId()}\" for {$class_storage->name}::"
+                                    . "{$const_name} does not satisfy the type "
+                                    . "\"{$parent_const_storage->type->getId()}\" inherited from "
+                                    . "{$parent_classlike_storage->name}::{$const_name}",
+                                $type_location,
+                                "{$class_storage->name}::{$const_name}"
+                            ),
+                            $const_storage->suppressed_issues
+                        );
+                    }
+                }
+
+                // Check overridden final
+                if ($parent_const_storage->final) {
                     IssueBuffer::maybeAdd(
-                        new LessSpecificClassConstantType(
-                            "The type '{$const_storage->type->getId()}' for {$class_storage->name}::"
-                                . "{$const_name} is more general than the type "
-                                . "'{$parent_const_storage->type->getId()}' inherited from "
-                                . "{$parent_classlike_storage->name}::{$const_name}",
-                            $location,
+                        new OverriddenFinalConstant(
+                            "{$const_name} cannot be overridden because it is marked as final in "
+                                . $parent_classlike_storage->name,
+                            $type_location,
                             "{$class_storage->name}::{$const_name}"
                         ),
                         $const_storage->suppressed_issues
                     );
                 }
             }
+
+            if ($const_storage->stmt_location !== null) {
+                // Check final in PHP < 8.1
+                if ($codebase->analysis_php_version_id < 8_01_00 && $const_storage->final) {
+                    IssueBuffer::maybeAdd(
+                        new ParseError(
+                            "Class constants cannot be marked final before PHP 8.1",
+                            $const_storage->stmt_location,
+                        ),
+                        $const_storage->suppressed_issues
+                    );
```

Comment:
Not sure if there's a better issue for this, I went with ParseError because that's what it is if you use some other invalid keyword.

Reply:
`ParseError` is good.